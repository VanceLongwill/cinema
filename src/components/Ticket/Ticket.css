.Ticket {
  position: relative;
  background: var(--secondary-dark-color);
  padding: var(--spacing-md);
  min-height: 150px;
  width: 100%;
  max-width: 500px;
  border-radius: 20px;
}

.TicketLeftEdge {
  position: absolute;
  display: flex;
  flex-direction: column;
  top: 50%;
  transform: translateY(-50%);
  left: -5px;
}

.LeftArrow {
	border-color: transparent var(--secondary-dark-color);
	border-style: solid;
	border-width: 10px 10px 10px 0px;
}

.TicketRightEdge {
  position: absolute;
  display: flex;
  flex-direction: column;
  top: 50%;
  transform: translateY(-50%);
  right: -5px;
}

.RightArrow {
	border-color: transparent var(--secondary-dark-color);
	border-style: solid;
	border-width: 10px 0px 10px 10px;
}

.TicketContent {
  border: 10px solid var(--secondary-light-color);
  border-radius: 20px;
  padding: var(--spacing-sm);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-weight: 700;
  font-size: 2rem;
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

