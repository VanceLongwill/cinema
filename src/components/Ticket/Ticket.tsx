import React from "react";

import "./Ticket.css";

export interface TicketProps {
  price: string;
  seatNumber: string;
}

export const Ticket: React.FunctionComponent<TicketProps> = ({
  price,
  seatNumber
}) => (
  <div className="Ticket">
    <div className="TicketLeftEdge">
      <span className="LeftArrow" />
      <span className="LeftArrow" />
      <span className="LeftArrow" />
      <span className="LeftArrow" />
      <span className="LeftArrow" />
      <span className="LeftArrow" />
      <span className="LeftArrow" />
    </div>
    <div className="TicketContent">
      <span>ADMIT ONE</span>
      <span>PRICE: {price}</span>
      <span>SEAT NO:{seatNumber}</span>
    </div>
    <div className="TicketRightEdge">
      <span className="RightArrow" />
      <span className="RightArrow" />
      <span className="RightArrow" />
      <span className="RightArrow" />
      <span className="RightArrow" />
      <span className="RightArrow" />
      <span className="RightArrow" />
    </div>
  </div>
);
