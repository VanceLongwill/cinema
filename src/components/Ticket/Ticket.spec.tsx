import * as React from "react"

import { mount } from "enzyme"

import { Ticket } from "./Ticket"

describe("Ticket", () => {
  it("should render without errors", () => {
    expect(mount(<Ticket />)).toMatchSnapshot()
  })
})
