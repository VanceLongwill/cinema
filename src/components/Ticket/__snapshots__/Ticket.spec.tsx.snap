// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket should render without errors 1`] = `
<Ticket>
  <div
    className="Ticket"
  >
    <div
      className="TicketLeftEdge"
    >
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
      <span
        className="LeftArrow"
      />
    </div>
    <div
      className="TicketContent"
    >
      <span>
        ADMIT ONE
      </span>
      <span>
        PRICE: 
      </span>
      <span>
        SEAT NO:
      </span>
    </div>
    <div
      className="TicketRightEdge"
    >
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
      <span
        className="RightArrow"
      />
    </div>
  </div>
</Ticket>
`;
